core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2073,6 +2073,41 @@ package object config {
       .timeConf(TimeUnit.MILLISECONDS)
       .createOptional
 
+  private[spark] val SPECULATION_EFFICIENCY_ENABLE =
+    ConfigBuilder("spark.speculation.efficiency.enabled")
+      .doc("When set to true, spark will evaluate the efficiency of task processing through the " +
+        "stage task metrics and only need to speculate the inefficient tasks.")
```

Comment:
nit: could also explain a bit more about what is an inefficient task? e.g., "A task is inefficient when its data process rate is less than the average data process rate of all successful tasks in the stage." Something like this.

Reply:
Ok I see you also mentioned at `spark.speculation.efficiency.progress.multiplier`. But this's the feature control flag so I think it's also needed to mention it here as well.